### PR TITLE
Data source for supported DBR/Spark versions

### LINKED ISSUE #347: Data source for spark versions
Create data source for retrieving spark versions:

```
data "databricks_spark_version" "latest" {
    latest = true # default
    long_term_support = false # default
    ml = true # default - false
    gpu = true # default - false
    scala = "2.12" # default
}
```

### PR DESCRIPTION
**Pull request by alexott**
This PR implements `databricks_spark_version` data source that allows to find DBR/Spark version that matches to specified filtering criteria.

This fixes #347

### CODE REVIEW

**Comment by alexott** [comment]
Please note that this doesn't make changes in the acceptance tests yet - only documentation was updated.  Modification of tests will be in the next commit.

File: compute/clusters.go [hunk 1]
```diff
@@ -388,3 +389,55 @@ func (a ClustersAPI) GetSmallestNodeType(r NodeTypeRequest) string {
...
+			fmt.Printf("versions=%v", versions)
+			return "", fmt.Errorf("Spark versions query returned multiple results. Please change your search criteria and try again")
+		}
+	}
+
+	return versions[0], nil
+}
+
+// Latest returns latest version matching the request parameters
+func (a ClustersAPI) Latest(svr SparkVersionRequest) (string, error) {
```
Comment (nfx) [inline compute/clusters.go:437]: this might be `LatestSparkVersion`?...

File: compute/data_spark_version.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/databrickslabs/databricks-terraform/internal"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// DataSourceSparkVersion returns DBR version matching to the specification
+func DataSourceSparkVersion() *schema.Resource {
+	s := internal.StructToSchema(SparkVersionRequest{}, func(
+		s map[string]*schema.Schema) map[string]*schema.Schema {
+		// s["scala"].ValidateFunc = validation.StringIsNotEmpty
```
Comment (nfx) [inline compute/data_spark_version.go:16]: can you remove commented code? :)

File: compute/clusters.go [hunk 3]
```diff
@@ -388,3 +389,55 @@ func (a ClustersAPI) GetSmallestNodeType(r NodeTypeRequest) string {
...
+			}
+		}
+	}
+	if len(versions) < 1 {
+		return "", fmt.Errorf("Spark versions query returned no results. Please change your search criteria and try again")
+	} else if len(versions) > 1 {
+		if req.Latest {
+			sort.Sort(sort.Reverse(sort.StringSlice(versions)))
+		} else {
+			fmt.Printf("versions=%v", versions)
```
Comment (nfx) [inline compute/clusters.go:428]: can you remove the string or add `[TRACE] ` in the beginning?

File: compute/data_spark_version.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		return s
+	})
+
+	log.Printf("[INFO] schema=%v", s)
```
Comment (nfx) [inline compute/data_spark_version.go:20]: this is more like `[TRACE]` level

File: compute/data_spark_version_test.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+func TestSparkVersionLight(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"light": true,
```
Comment (nfx) [inline compute/data_spark_version_test.go:122]: i don't know if we should even include light :D this project should drive DBUs as much as possible and `light` is never recommended.

File: compute/data_spark_version_test.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"latest": false,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.Error(t, err)
+	assert.Equal(t, true, strings.Contains(err.Error(), "query returned multiple results"))
```
Comment (nfx) [inline compute/data_spark_version_test.go:189]: shouldn't the query do latest by default?

File: compute/model.go [hunk 7]
```diff
@@ -637,3 +637,28 @@ type EventsResponse struct {
...
+
+// SparkVersionRequest - filtering request
+type SparkVersionRequest struct {
+	LongTermSupport bool   `json:"long_term_support,omitempty" tf:"optional,default:false"`
+	Beta            bool   `json:"beta,omitempty" tf:"optional,default:false,conflicts:long_term_support"`
+	Latest          bool   `json:"latest,omitempty" tf:"optional,default:true"`
+	ML              bool   `json:"ml,omitempty" tf:"optional,default:false"`
+	Genomics        bool   `json:"genomics,omitempty" tf:"optional,default:false"`
+	GPU             bool   `json:"gpu,omitempty" tf:"optional,default:false"`
+	Light           bool   `json:"light,omitempty" tf:"optional,default:false"`
```
Comment (nfx) [inline compute/model.go:661]: let's remove light :D

**Review by nfx — CHANGES_REQUESTED** [review]
This is so awesome!

left few minor comments

File: compute/data_spark_version_test.go (hunk 6 above)
Comment (alexott) [reply compute/data_spark_version_test.go:189]: not in this case - it explicitly says `latest = false`, just to trigger error handling...

**Comment by alexott** [comment]
I'll update code tomorrow... And will try to update acceptance tests as well.

**Comment by alexott** [comment]
ok, all comments are addressed.  I think that working on acceptance tests in this branch doesn't make sense because most probably it will conflict with changes for tests that are required for #417

**Pull request merged**